lib/zlib.js

```diff
@@ -431,6 +431,11 @@ Zlib.prototype._processChunk = function _processChunk(chunk, flushFlag, cb) {
     if (self._hadError)
       return;
 
+    if (self.destroyed) {
+      this.buffer = null;
```

Comment:
I think there is no need to do this. It's done implicitly in https://github.com/Fishrock123/node/blob/9b2aee0fcea8eea690976a6ad3cb30b136ba6a67/lib/zlib.js#L426-L429